flybywiresim/a32nx
# feat(nd): track up mode

## Summary of Changes


Credit @tracernz for the implementation.

- [x] Introduces ND Track Up mode pin-program, enabled via the "Aircraft Configuration" section of the flyPad settings
- [x] Necessary map rotation and needle changes
- [x] TRK flag (when ADIRS not aligned for now; when track invalid later)

## Screenshots (if necessary)



![image](https://user-images.githubusercontent.com/4503241/143978484-e4e49d7b-3d9a-4a76-94f3-06213930c523.png)

![image](https://user-images.githubusercontent.com/4503241/143978349-d2b97f35-3e5e-47f2-be2e-ef06b6005e93.png)

![image](https://user-images.githubusercontent.com/4503241/143978368-cbd69417-ca8e-4c15-8b2b-cfefc0d36d49.png)

## References





https://www.pprune.org/archive/index.php/t-566053.html

FBW-31-03

## Additional context



Discord username (if different from GitHub):

## Testing instructions


- Make sure heading up mode on the ND is accurate in regards to:
  - aircraft symbol and lubber (yellow) line do **not** rotate
  - aircraft points towards heading
  - track is represented by the green diamond 
  - needles are accurate
  - LS bug is accurate
  - "TRK" flag is **not** shown when ADIRS unaligned
 
- Make sure track up mode on the ND is accurate in regards to (better to test with a crosswind):
  - aircraft symbol and lubber (yellow) line **do** rotate
  - aircraft points towards heading
  - track is represented by the green diamond, which never moves
  - needles are accurate
  - LS bug is accurate
  - "TRK" flag **is** shown when ADIRS unaligned


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
